lib/web/ui/integration.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ui
+
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// IntegrationAWSOIDCSpec contain the specific fields for the `aws-oidc` subkind integration.
+type IntegrationAWSOIDCSpec struct {
+	// RoleARN is the role associated with the integration when SubKind is `aws-oidc`
+	RoleARN string `json:"roleArn,omitempty"`
+}
+
+// Integration describes the base Integration fields (can be read or wri) Integration
+type Integration struct {
+	// Name is the Integration name.
+	Name string `json:"name"`
+	// SubKind is the Integration SubKind.
+	SubKind string `json:"subKind"`
+	// AWSOIDC contains the fields for `aws-oidc` subkind integration.
+	AWSOIDC *IntegrationAWSOIDCSpec `json:"awsoidc"`
+}
+
+// CheckAndSetDefaults for the create request.
+// Name and SubKind is required.
+func (r *Integration) CheckAndSetDefaults() error {
+	if r.Name == "" {
+		return trace.BadParameter("missing integration name")
+	}
+
+	if r.SubKind == "" {
+		return trace.BadParameter("missing subKind")
+	}
+
+	if r.AWSOIDC != nil && r.AWSOIDC.RoleARN == "" {
+		return trace.BadParameter("missing awsoidc.roleArn field")
+	}
+
+	return nil
+}
+
+// UpdateIntegrationRequest is a request to update an Integration
+type UpdateIntegrationRequest struct {
+	// AWSOIDC contains the fields for `aws-oidc` subkind integration.
+	AWSOIDC *IntegrationAWSOIDCSpec `json:"awsoidc"`
+}
+
+// CheckAndSetDefaults checks if the provided values are valid.
+func (r *UpdateIntegrationRequest) CheckAndSetDefaults() error {
+	if r.AWSOIDC != nil && r.AWSOIDC.RoleARN == "" {
+		return trace.BadParameter("missing awsoidc.roleArn field")
+	}
+
+	return nil
+}
+
+// IntegrationsListResponse contains a list of Integrations.
+// In case of exceeding the pagination limit (either via query param `limit` or the default 1000)
+// a `nextToken` is provided and should be used to obtain the next page (as a query param `startKey`)
+type IntegrationsListResponse struct {
+	// Items is a list of resources retrieved.
+	Items interface{} `json:"items"`
```

Comment:
are we expecting different types of integration in this list?
```suggestion
	Items []Integration `json:"items"`
```